include/pmacc/Environment.hpp

```diff
@@ -497,12 +498,17 @@ namespace detail
                      * - to set the flags reset the device and set flags again
                      */
                     CUDA_CHECK(cudaDeviceReset());
-                    CUDA_CHECK(cudaSetDeviceFlags(cudaDeviceScheduleSpin));
+                    CUDA_CHECK((cuplaError_t)cudaSetDeviceFlags(cudaDeviceScheduleSpin));
                 }
+#endif
                 CUDA_CHECK(cudaGetLastError());
                 break;
             }
-            else if (rc == cudaErrorDeviceAlreadyInUse || rc==cudaErrorDevicesUnavailable)
+            else if (rc == cudaErrorDeviceAlreadyInUse
+#if (PMACC_CUDA_ENABLED == 1)
+                || rc==(cudaError)cudaErrorDevicesUnavailable
```

Comment:
must be `cudaError` and will be changed in a follow up pull request if we switch fully to cupla natives.